launcher/src/main/java/org/apache/spark/launcher/LauncherServer.java

```diff
@@ -260,24 +264,30 @@ private long getConnectionTimeout() {
   }
 
   private String createSecret() {
-    byte[] secret = new byte[128];
-    RND.nextBytes(secret);
-
-    StringBuilder sb = new StringBuilder();
-    for (byte b : secret) {
-      int ival = b >= 0 ? b : Byte.MAX_VALUE - b;
-      if (ival < 0x10) {
-        sb.append("0");
+    while (true) {
```

Comment:
checking my understanding -- extra bug fix here?  even in old code, if by chance two apps had same secret, you'd end up losing one handle?

Reply:
Not really, it's mostly moving the logic that existed before (look for `while (server.pending.containsKey(secret))`).